clippy_lints/src/redundant_clone.rs

```diff
@@ -208,8 +208,9 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for RedundantClone {
 
             if !used_later {
                 let span = terminator.source_info.span;
-                let node = if let mir::ClearCrossCrate::Set(scope_local_data) = &mir.source_scope_local_data {
-                    scope_local_data[terminator.source_info.scope].lint_root
+                let scope = terminator.source_info.scope;
+                let node = if let mir::ClearCrossCrate::Set(scope_local_data) = &mir.source_scopes[scope].local_data {
+                    scope_local_data.lint_root
                 } else {
                     unreachable!()
```

Comment:
If you're going to panic, please use `.as_ref().assert_crate_local()` like I did in my PR.

Reply:
That makes sense, thanks!